Document how to declare a PostCSS plugin in an ES6/TypeScript module

It's currently unclear how to properly declare a PostCSS module using ES6 module syntax (which TypeScript also uses idiomatically). The suggested `module.exports` isn't idiomatic in these systems.

That was the first thing I tried, but it doesn't work if you're loading it from TypeScript:

```
 FAIL  test/index.test.ts
  ● Test suite failed to run

    test/index.test.ts:22:25 - error TS2349: This expression is not callable.
      Type 'typeof import("/home/nweiz/goog/pkg/js/postcss-src/build/index")' has no call signatures.

    22   return await postcss([plugin(options)]).process(input, {from: undefined});
                               ~~~~~~
```

I also tried

```ts
const plugin: PluginCreator = (opts: Options = {}): Plugin => { … };
plugin.postcss = true;

export = plugin;
```

but this produced

```
src/index.ts:80:1 - error TS2309: An export assignment cannot be used in a module with other exported elements.                       

80 export = plugin;
   ~~~~~~~~~~~~~~~~
```

So far, this is the best I've been able to manage:

```ts
const plugin: PluginCreator = (opts: Options = {}): Plugin => { … };

export = plugin;
```

...but this doesn't include the `postcss` flag, which I assume is desirable.

--- a/docs/guidelines/plugin.md
+++ b/docs/guidelines/plugin.md
@@ -119,6 +119,10 @@ module.exports = opts => {
 module.exports.postcss = true
 ```
 
+For ESM / TypeScript modules, use the same `postcss = true` flag on the
+creator function and `export default plugin` (see
+[Writing a PostCSS Plugin](../writing-a-plugin.md)).
+
 ## 2. Processing
 
 ### 2.1. Plugin must be tested

--- a/docs/guidelines/plugin.md
+++ b/docs/guidelines/plugin.md
@@ -108,7 +108,7 @@ It is better even not to import `postcss`.
 Plugin name will be used in error messages and warnings.
 
 ```js
-module.exports = opts => {
+const plugin = opts => {
   return {
     postcssPlugin: 'postcss-name',
     Once(root) {
@@ -116,12 +116,10 @@ module.exports = opts => {
     }
   }
 }
-module.exports.postcss = true
-```
+plugin.postcss = true
 
-For ESM / TypeScript modules, use the same `postcss = true` flag on the
-creator function and `export default plugin` (see
-[Writing a PostCSS Plugin](../writing-a-plugin.md)).
+export default plugin
+```
 
 ## 2. Processing
 

--- a/docs/writing-a-plugin.md
+++ b/docs/writing-a-plugin.md
@@ -101,6 +101,37 @@ module.exports = (opts = {}) => {
 module.exports.postcss = true
 ```
 
+For ESM, attach `postcss = true` to the creator function and default-export it:
+
+```js
+const plugin = (opts = {}) => {
+  // Plugin creator to check options or prepare shared state
+  return {
+    postcssPlugin: 'PLUGIN NAME'
+    // Plugin listeners
+  }
+}
+plugin.postcss = true
+
+export default plugin
+```
+
+In TypeScript, type the creator as `PluginCreator` so `postcss = true` is valid:
+
+```ts
+import type { PluginCreator } from 'postcss'
+
+const plugin: PluginCreator<object> = (opts = {}) => {
+  return {
+    postcssPlugin: 'PLUGIN NAME'
+    // Plugin listeners
+  }
+}
+plugin.postcss = true
+
+export default plugin
+```
+
 [PostCSS plugin boilerplate]: https://github.com/postcss/postcss-plugin-boilerplate/
 [plugin template]: https://github.com/postcss/postcss-plugin-boilerplate/blob/main/template/index.t.js
 

--- a/docs/writing-a-plugin.md
+++ b/docs/writing-a-plugin.md
@@ -91,19 +91,6 @@ For public plugins:
 3. Publish your code there.
 
 ```js
-module.exports = (opts = {}) => {
-  // Plugin creator to check options or prepare shared state
-  return {
-    postcssPlugin: 'PLUGIN NAME'
-    // Plugin listeners
-  }
-}
-module.exports.postcss = true
-```
-
-For ESM, attach `postcss = true` to the creator function and default-export it:
-
-```js
 const plugin = (opts = {}) => {
   // Plugin creator to check options or prepare shared state
   return {

